flocker/control/httpapi.py

```diff
@@ -492,11 +498,15 @@ def create_container_configuration(self, host, name, image, ports=()):
                 external_port=port['external']
             ))
 
+        if environment:
```

Comment:
Why have this check at all then?  It looks like it can only introduce complexity - and an extra bug.  If `environment` is `{}` then it won't get converted to a frozenset, so it's passed to `Application` as a dict.  Maybe that doesn't break anything, but why go out of our way to do it?